#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From aa75395f2d4b203b3b764f2a3a622dc4f465e930 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@intevation.de>
Date: Fri, 19 Feb 2016 14:57:57 +0100
Subject: [PATCH] Use non native Language name as fallback

Some languages like Lower German don't have a native Language
name in Qt. So we use a fallback to avoid showing empty menu
entries.
---
 src/kswitchlanguagedialog_p.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/kswitchlanguagedialog_p.cpp b/src/kswitchlanguagedialog_p.cpp
index 039daea..531598e 100644
--- a/src/kswitchlanguagedialog_p.cpp
+++ b/src/kswitchlanguagedialog_p.cpp
@@ -339,12 +339,16 @@ void KSwitchLanguageDialogPrivate::fillApplicationLanguages(KLanguageButton *but
     foreach(const QLocale &l, allLocales) {
         QString languageCode = l.name();
         if (l != cLocale) {
+            const QString nativeName = l.nativeLanguageName();
+            // For some languages the native name is empty. In this case use the non native language
+            // name as fallback.
+            const QString languageName = nativeName.isEmpty() ? QLocale::languageToString(l.language()) : nativeName;
             if (!insertedLanguges.contains(languageCode) && KLocalizedString::isApplicationTranslatedInto(languageCode)) {
-                button->insertLanguage(languageCode, l.nativeLanguageName());
+                button->insertLanguage(languageCode, languageName);
                 insertedLanguges << languageCode;
             } else if (stripCountryCode(&languageCode)) {
                 if (!insertedLanguges.contains(languageCode) && KLocalizedString::isApplicationTranslatedInto(languageCode)) {
-                    button->insertLanguage(languageCode, l.nativeLanguageName());
+                    button->insertLanguage(languageCode, languageName);
                     insertedLanguges << languageCode;
                 }
             }
--
2.1.4
